src/fielddef.ts

```diff
@@ -156,6 +156,8 @@ export function field(fieldDef: FieldDef, opt: FieldRefOption = {}) {
 
   if (opt.datum) {
     field = `datum["${field}"]`;
+  } else if (field !== undefined && field.indexOf('.') !== -1) { // dot in fieldname
```

Comment:
@domoritz  Hmm, but by doing what you suggest would support only one thing if we stick with what Vega already does for us we support both